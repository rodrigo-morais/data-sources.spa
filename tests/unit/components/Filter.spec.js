import Vuex from 'vuex'
import { mount, createLocalVue } from '@vue/test-utils'
import { createRenderer } from 'vue-server-renderer'
import SearchFilter from '@/components/SearchArea/Filter/index.vue'

const localVue = createLocalVue()
localVue.use(Vuex)

const actions = { fetchDataSources: jest.fn() }
const state = {
  loading: false,
  data: [{ name: 'DataSource 1' }, { name: 'DataSource 2' }],
  error: false,
}
const store = new Vuex.Store({
  state,
  actions,
})

describe('SearchFilter.vue', () => {
  let wrapper

  beforeEach(() => {
    wrapper = mount(SearchFilter, { store, localVue })
    actions.fetchDataSources.mockReset()
  })

  it('renders according to design', () => {
    const renderer = createRenderer()

    renderer.renderToString(wrapper.vm, (err, str) => {
      if (err) throw new Error(err)
      expect(str).toMatchSnapshot()
    })
  })

  it('dispatches "fetchDataSources" when TextBox has more than 3 characters', () => {
    const input = wrapper.find('#searchBox')
    input.element.value = 'data'
    input.trigger('input')

    expect(actions.fetchDataSources).toHaveBeenCalled()
  })

  it('dispatches "fetchDataSources" when Checkbox is selected', () => {
    const checkbox = wrapper.find('.checkbox')
    checkbox.trigger('click')

    expect(actions.fetchDataSources).toHaveBeenCalled()
  })

  it('dispatches "fetchDataSources" when Dropdown is selected', () => {
    const dropdownItem = wrapper.find('.dropdown__menu').find('ul').find('li')
    dropdownItem.trigger('click')

    expect(actions.fetchDataSources).toHaveBeenCalled()
  })
})
