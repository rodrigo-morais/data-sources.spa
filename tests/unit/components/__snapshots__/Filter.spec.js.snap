// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchFilter.vue renders according to design 1`] = `
<div class="filter">
  <SearchTextBox-stub></SearchTextBox-stub>
  <Dropdown-stub className="dropdown__legal-entity" label="Legal Entity" items="Show All,Yes,No"></Dropdown-stub>
  <Dropdown-stub className="dropdown__sort-by" label="Sort by" items="Most recent,Oldest"></Dropdown-stub>
  <Checkbox-stub className="checkbox__archived" label="Show archived" isChecked="true"></Checkbox-stub>
</div>
`;
